src/Suggestor/EditorSuggestorPopup.ts

```diff
@@ -14,6 +14,15 @@ export class EditorSuggestor extends EditorSuggest<SuggestInfoWithContext> {
     constructor(app: App, settings: Settings) {
         super(app);
         this.settings = settings;
+
+        // EditorSuggestor swallows tabs while the suggestor popup is open
+        // This is a hack to support indenting while popup is open
+        app.scope.register([], 'Tab', () => {
+            const editor = this.context?.editor;
+            if (editor) {
+                editor.replaceRange('\t', { line: editor.getCursor().line, ch: 0 });
```

Comment:
The options to that call are somewhat cryptic to my untutored eye.

I've read it several times and I have no idea how the `replaceRange()` call fixes the issue.

An explanatory comment would be appreciated.

Reply:
I've added a clarifying comment above this line. replaceRange basically functions as an insert if you don't pass anything for the `to` argument:

```typescript
replaceRange(replacement: string, from: EditorPosition, to?: EditorPosition, origin?: string): void;
```

So we're just inserting a tab character at the 0th char position on the current line - hope that makes it clear!

Reply:
> So we're just inserting a tab character at the 0th char position on the current line - hope that makes it clear!

Thanks... And presumably if the user setting is to use spaces instead of tabs, Obsidian will convert that tab character to the correct number of spaces?

Reply:
Ah good spot, I hadn't considered this setting & it doesn't get converted. I'll look into whether we can access global settings and insert spaces or tabs accordingly

Reply:
Thanks for the suggestion...
Hmmm. Now it's getting quite a complicated fix.

Reply:
I wonder if there is anything in the Obsidian API or the code itself to indent text one level? It might take care of the settings.

Reply:
Or maybe Ush or someone else on the forum ticket might have a suggestion for a way without having to figure out the settings. I mean, it is possible to read them, but I suspect if we do, we will get in to all sorts of corner cases and end up rewriting bits of Obsidian.

Reply:
I found `editor.exec('indentMore')` in the end which does what we want :)

Also added `preventDefault` behaviour on Ush's suggestion which *should* nullify any concerns about an upstream fix causing two indents (hard to test that case til it happens though)

Reply:
I've re-tested all of the test cases you mentioned and it looks good. Let me know if any other edge cases occur to you, else I'm happy for this to be merged!

Reply:
Thanks, it looks really good.

I'm unlikely to be able to test it for a few days, so (as you will have seen) I've asked for help with testing in #1578.